packages/mongodb-chatbot-server/src/routes/responses/createResponse.ts

```diff
@@ -87,8 +95,30 @@ const InputMessageSchema = z.object({
   ]),
 });
 
+const AssistantMessageSchema = z.object({
+  type: z.literal("message").optional(),
+  role: z.literal("assistant"),
+  content: z.union([
+    z.string(),
+    z
+      .array(
+        z.object({
+          type: z.literal("output_text"),
+          text: z.string(),
+        })
+      )
+      .length(1, CREATE_RESPONSE_ERR_MSG.INPUT_ASSISTANT_CONTENT_ARRAY),
+  ]),
+});
+
+const InputMessageSchema = z.union([
+  SystemMessageSchema,
+  UserMessageSchema,
+  AssistantMessageSchema,
+]);
+
 type InputMessage = z.infer<typeof InputMessageSchema>;
-type UserMessage = Omit<InputMessage, "role"> & { role: "user" };
+type UserMessage = z.infer<typeof UserMessageSchema>;
 
 const CreateResponseRequestBodySchema = z.object({
   model: z.string(),
```

Comment:
i think would be cleaner, more readable code if the 'function tool call' and 'function tool call output' schemas (starting line 140) were each made their own variable and then added to the InputMessageSchema above.